pkg/operator/standalone_starter.go

```diff
@@ -33,6 +35,31 @@ func startControllerStandAlone(ctx context.Context, controllerConfig *controller
 	versionGetter := status.NewVersionGetter()
 	versionGetter.SetVersion("operator", status.VersionForOperatorFromEnv())
 
+	desiredVersion := status.VersionForOperatorFromEnv()
+	missingVersion := "0.0.1-snapshot"
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		clients.ConfigInformers.Config().V1().ClusterVersions(), clients.ConfigInformers.Config().V1().FeatureGates(),
+		controllerConfig.EventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go clients.ConfigInformers.Start(ctx.Done())
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		klog.Infof("FeatureGates initialized: knownFeatureGates=%v", featureGates.KnownFeatures())
+	case <-time.After(1 * time.Minute):
+		klog.Errorf("timed out waiting for FeatureGate detection")
+		return fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+	featureGates, err := featureGateAccessor.CurrentFeatureGates()
```

Comment:
I feel like `CurrentFeatureGates` should block until it has done syncing the loop. Is 1 minute reasonable time btw?

Reply:
The select above is the blocking sync to get feature gates. 

With `exit()` after 1 minute, kubelet will start exp. backoff restarting the Pod, so I think it's good enough. A cluster without FeatureGate is not a very healthy one.

Reply:
@deads2k what do you think about call to `CurrentFeatures` blocking . I think that this code leaks internal details of  `featuregateAccess`  object.  We can clean this up in a follow up, if we all agree to it.